js/how-does-javascript-work/how-does-javascript-work.md

```diff
@@ -0,0 +1,149 @@
+It executes the first line `var n = 2` and replaces the `undefined` value of `n` from the Memory block with the actual value `2`.
+
+Then it comes to line 2 and encounters a function `square` and skips the code from line 2 to line 5.
+
+Then when it goes to line 6 and encounters a function call or a function invocation.
+Now it knows that it has to execute a function.
+
+> " Functions are the heart of JavaScript " - Akshay Saini
+
+Whenever a function is invoked a new `Execution Context` will be created and this will be a brand new `Functional Execution Context(FEC)` inside the `Global Execution Context(GEC)`
+
+This new `Execution Context` will again consist of two blocks, **Memory** and **Code**, and will also follow the two phases of creation.
+
+In its **Memory Creation phase**, it will allocate memory to variables and functions used in its execution.
+
+It scans for variables and it finds `num` as a parameter and `ans` in the body and assigns `undefined` to them.
+
+In the **Code Execution phase** the value of `n` which is 2 is passed to `num`.
+
+Now in the function body, it executes `num * num` and the evaluated value is replaced for `undefined` in `ans` inside the `FEC`.
+
+After execution of line 3, it encounters a `return` keyword on line 4, so the function knows its execution is finished and now it's time to return the control back to the `Execution Context` from where it was invoked which is the `GEC`.
+
+While returning it returns the value of `ans` from its `Execution Context` to the Global one so the value of `square2` is changed from `undefined` to `4`.
+
+**The whole `Functional Execution Context` is deleted after the return of control.**
+
+A visual representation of the whole process discussed: 
+
+![FEC1.png](https://cdn.hashnode.com/res/hashnode/image/upload/v1611861952440/MPoJltLlU.png)
+
+Now the control moves to line 7, where it finds another function invocation `square(4)` with value 4 as an argument. Again brand new `FEC` will be created and the whole procedure used to invoke `square(n)` as explained above will be repeated and the value will be evaluated and returned into square4.
+
+If there is a function inside a function then a nested `FEC` will be created inside the `FEC` which is inside the `GEC`. This has no end it can go beyond and beyond.
+
+After the execution of the whole program is finished, The `Global Execution Context` also gets deleted.
+
+Phew, that was all about the **"Execution Context"**, now you know how is it created and how it executes the code which we write in JavaScript.
+
+Remember I said all this nested `Execution Context` can go on and on, so how does JavaScript handle all this complicated nesting and control of the program.
+Luckily JavaScript engine has a stack to manage all these things and we call it 
+**The Call Stack**, let's learn more about it in detail.
+
+### 2. Call Stack
+You must be familiar with the stack data structure. If not please read about it.
```

Comment:
Instead of saying "If not please read about it.", can we say "If not, please read it here." with a link to a site which explains the topic. This will be really helpful to readers.